From 7f06b67b7624a771799913c8a4dc031f94062c29 Mon Sep 17 00:00:00 2001
From: Simon Cook <simon.cook@embecosm.com>
Date: Tue, 23 Sep 2014 15:53:07 +0100
Subject: [PATCH 15/33] * config/epiphany/epiphany.h (ENDFILE_SPEC): Add
 cachemanager.o when -fsoftware-cache is used. (CC1_SPEC): Add -fpic when
 -fsoftware-cache is used. * config/epiphany/epiphany.opt: New option.

---
 gcc/ChangeLog.epiphany           | 7 +++++++
 gcc/config/epiphany/epiphany.h   | 4 +++-
 gcc/config/epiphany/epiphany.opt | 3 +++
 3 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index c2d9f059567..f66a52b0ff4 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,10 @@
+2014-09-23  Simon Cook  <simon.cook@embecosm.com>
+
+	* config/epiphany/epiphany.h (ENDFILE_SPEC): Add cachemanager.o
+	when -fsoftware-cache is used.
+	(CC1_SPEC): Add -fpic when -fsoftware-cache is used.
+	* config/epiphany/epiphany.opt: New option.
+
 2014-05-30  Pierre Langlois  <pierre.langlois@embecosm.com>
 
 	* config/epiphany/epiphany.h (EPIPHANY_LIBRARY_EXTRA_SPEC): Build
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index 7f0706047cf..3a63160b53e 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -55,7 +55,9 @@ along with GCC; see the file COPYING3.  If not see
   "%{m1reg-r43:crtm1reg-r43.o%s} %{m1reg-r63:crtm1reg-r63.o%s} " \
   "crtbegin.o%s"
 
-#define ENDFILE_SPEC "crtend.o%s crtn.o%s"
+#define ENDFILE_SPEC "crtend.o%s crtn.o%s %{fsoftware-cache:cachemanager.o%s}"
+
+#define CC1_SPEC "%{fsoftware-cache:-fpic}"
 
 #define EPIPHANY_LIBRARY_EXTRA_SPEC \
   "-ffixed-r41 -ffixed-r41 -ffixed-r42 -ffixed-r43 -fpic"
diff --git a/gcc/config/epiphany/epiphany.opt b/gcc/config/epiphany/epiphany.opt
index dd194a684d0..9055f0e62c7 100644
--- a/gcc/config/epiphany/epiphany.opt
+++ b/gcc/config/epiphany/epiphany.opt
@@ -146,3 +146,6 @@ Enum(m1reg) String(r43) Value(43)
 
 EnumValue
 Enum(m1reg) String(r63) Value(63)
+
+fsoftware-cache
+Use software cache
-- 
2.31.1

